src/ripple/protocol/impl/Feature.cpp

```diff
@@ -27,67 +28,87 @@
 
 namespace ripple {
 
-static boost::container::flat_map<uint256, std::size_t> featureToIndex;
-static std::vector<uint256> features;
-
-size_t featureToBitsetIndex(uint256 f)
-{
-    auto const i = featureToIndex.find(f);
-    assert(i != featureToIndex.end());
-    return i->second;
-}
-
-uint256 bitsetIndexToFeature(size_t i)
-{
-    assert(i < features.size());
-    return features[i];
-}
-
 static
 uint256
 feature (char const* s, std::size_t n)
 {
     sha512_half_hasher h;
     h(s, n);
-    auto const result = static_cast<uint256>(h);
-    if (featureToIndex.find(result) == featureToIndex.end())
-    {
-        featureToIndex[result] = features.size();
-        features.push_back(result);
-        assert(features.size() < FeatureBitsetSize); // Need to increase FeatureBitsetSize
-    }
-    return result;
+    return static_cast<uint256>(h);
 }
 
+static
 uint256
 feature (std::string const& name)
 {
     return feature(name.c_str(), name.size());
 }
 
-uint256
-feature (const char* name)
+static std::vector<uint256> const features = [] {
+    std::vector<uint256> result;
+    result.reserve(detail::numFeatures);
+    for (std::string const& s : detail::featureNames)
+        result.push_back(feature(s));
+    return result;
+}();
+
+static boost::container::flat_map<uint256, std::size_t> const featureToIndex =
+    [] {
+        boost::container::flat_map<uint256, std::size_t> result;
+        result.reserve(detail::numFeatures);
+        size_t i = 0;
+        for (std::string const& s : detail::featureNames)
+            result[feature(s)] = i++;
+        return result;
+    }();
+
+static boost::container::flat_map<std::string, uint256> const nameToFeature = []{
+    boost::container::flat_map<std::string, uint256> result;
+    result.reserve(detail::numFeatures);
+    for (std::string const& s : detail::featureNames)
+        result[s] = feature(s);
+    return result;
+}();
```

Comment:
I'll note that structured bindings would be useful here when we move to `17. Each of the initializers to `features`, `featureToIndex` and `nameToFeature` are similar and it would make sense to put them all in the same function. However, given we want to make these collections `const` - and in this case I really care about that `const` - I don't see a good way to do that. I _think_ something like `auto const [features, featuresToIndex, nameToFeatuers] = ...` would work well.